mp2-common/src/eth.rs

```diff
@@ -111,6 +116,175 @@ pub struct ProofQuery {
+    pub tx_index: u64,
+    /// The size of the index in bytes
+    pub index_size: usize,
+    /// The offset in the leaf (in RLP form) to status
+    pub status_offset: usize,
+    /// The offset in the leaf (in RLP form) to the start of logs
+    pub logs_offset: usize,
+    /// Data about the type of log we are proving the existence of
+    pub event_log_info: EventLogInfo,
+    /// The offsets for the relevant logs
+    pub relevant_logs_offset: Vec<usize>,
+}
+
+/// Contains all the information for an [`Event`] in rlp form
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct EventLogInfo {
+    /// Size in bytes of the whole log rlp encoded
+    pub size: usize,
+    /// Packed contract address to check
+    pub address: Address,
+    /// Byte offset for the address from the beginning of a Log
+    pub add_rel_offset: usize,
+    /// Packed event signature,
+    pub event_signature: [u8; 32],
+    /// Byte offset from the start of the log to event signature
+    pub sig_rel_offset: usize,
+    /// The topics for this Log
+    pub topics: [LogDataInfo; 3],
+    /// The extra data stored by this Log
+    pub data: [LogDataInfo; 2],
+}
+
+/// Contains all the information for data contained in an [`Event`]
+#[derive(Debug, Clone, Copy, Default, Serialize, Deserialize)]
+pub struct LogDataInfo {
+    pub column_id: usize,
+    /// The byte offset from the beggining of the log to this target
+    pub rel_byte_offset: usize,
+    /// The length of this topic/data
+    pub len: usize,
+}
+
+impl TryFrom<&Log<LogData>> for EventLogInfo {
+    type Error = anyhow::Error;
+
+    fn try_from(log: &Log<LogData>) -> std::result::Result<Self, Self::Error> {
+        // First we encode the log in rlp form
+        let mut buf = Vec::<u8>::new();
+        log.encode(&mut buf);
+
+        let rlp_log = rlp::Rlp::new(&buf);
+        // Extract the header
+        let log_header = rlp_log.payload_info()?;
+        let next_data = &buf[log_header.header_len..log_header.header_len + log_header.value_len];
+        let rlp_log_no_header = rlp::Rlp::new(next_data);
+        // Find the address offset (skipping its header)
+        let address_header = rlp_log_no_header.payload_info()?;
+        let rel_address_offset = log_header.header_len + address_header.header_len;
+        // Find the signature offset (skipping its header)
+        let topics_data = &buf[rel_address_offset + address_header.value_len
+            ..log_header.header_len + log_header.value_len];
+        let topics_rlp = rlp::Rlp::new(topics_data);
+        let topics_header = topics_rlp.payload_info()?;
+        let topic_0_data =
+            &buf[rel_address_offset + address_header.value_len + topics_header.header_len
+                ..log_header.header_len
+                    + address_header.header_len
+                    + address_header.value_len
+                    + topics_header.header_len
+                    + topics_header.value_len];
```

Comment:
can you keep a increasing index ? feels clearer to me than repeating all the time what is in front, a lot of lines !